### ISSUE TITLE
BYOL loss function

### ISSUE THREAD

**Issue by vishnu-dev** [Lightning-Universe/lightning-bolts#1058]
## 🐛 Bug

The loss function in BYOL doesn't seem to match the one defined in the paper.

### To Reproduce

Running the training for BYOL.

#### Code sample

https://github.com/Lightning-Universe/lightning-bolts/blob/4f910f6c9893eaf206eed6b7df545c8a02043a55/src/pl_bolts/models/self_supervised/byol/byol_module.py#L130-L142

### Expected behavior

The loss should be `2 - (2 * F.cosine_similarity(h1, z2).mean())`?

<img width="717" alt="Screenshot 2023-07-16 at 3 22 50 PM" src="https://github.com/Lightning-Universe/lightning-bolts/assets/21088161/d11e18b2-e1cc-4aa1-af1b-7be089d84160">

**Comment by Borda** [comment]
Thank you, @vishnu-dev mind sending PR with the fix? :rabbit:
